### PR TITLE
BinaryOperatorSpacesFixer - Single space minimum option

### PR DESCRIPTION
**Pull request by stevelacey**
Assuming this is of interest I'll tidy it up as per the contribution guide.

This allows a config like this:

```
        'binary_operator_spaces' => [
            'default' => 'single_space',
            'operators' => [
                '=' => 'single_space_minimum',
                '=>' => 'single_space_minimum',
            ]
        ],
```

Which allows a less drastic change than either forcing single spaces or forcing alignment for all `=` and `=>` but still enforcing at least one space – which are currently the only available options afaics.

The naming of this is tricky, I think the other names are a bit confusing, I also thought about calling this `single_space_or_more` to distinguish it from `align_single_space_minimal`. Is that better?

An alternative (or extra) implementation I'd probably prefer would be `single_space_or_align`... but that'd be a lot harder to throw together than this was.

For comparison, on my current project, using `single_space` or `align_single_space_minimal` resulted in 300+ file changes, given there's a whole mix of legitimately aligned and single-spaced things, whilst `single_space_minimum` caused 5 file changes, sniping out the specific problem I wanted to address: zero spaces.

Thoughts? Shall I tidy this up and add the docs and tests?

Related: https://github.com/FriendsOfPHP/PHP-CS-Fixer/issues/4396

### CODE REVIEW

**Comment by stevelacey** [comment]
@SpacePossum @keradus ?

**Comment by stevelacey** [comment]
@julienfalque @nicolas-grekas?

**Comment by nicolas-grekas** [comment]
No opinion on my side as I don't think I'd use this rule. Single space is my all time favorite :)

**Comment by SpacePossum** [comment]
Hi and apologies for getting back so late.

Am I correct to understand you want something like this:

```php
$a =$b&&$c; // fix
$d =$a   &&    $e; // don't fix
```

being fixed to this:

```php
$a =$b && $c; // fix
$d =$a   &&    $e; //don't fix
```

? 
It might help having unit tests as an example :)

Im +0 on adding the option, but leave the remark that Im very much in favor to rename the options in 3.0.0 because there is a lot of confusion we could get rid of :+1:

**Comment by stevelacey** [comment]
@SpacePossum yes, whilst those examples aren't really what I'm after, that's what the effect would be, in short, I want to enforce a minimum of 1 space, without enforcing or disallowing alignment.

If it were trivial to write "single space or alignment" as a rule I'd do that, but that's pretty complex.

I am happy to spend more time on this and submit update this PR with tests... if it'd ever merge.

**Comment by SpacePossum** [comment]
>if it'd ever merge.

I cant promise that as we would have to see what others think as well, but I would vote for it :)
There are a few details to sort out though, not sure if you want to work on this or if you want to wait for more people expressing support for your PR in concept?

File: src/Fixer/Operator/BinaryOperatorSpacesFixer.php [hunk 1]
```diff
@@ -393,6 +405,33 @@ private function fixWhiteSpaceAroundOperatorToSingleSpace(Tokens $tokens, $index
...
+    /**
+     * @param Tokens $tokens
+     * @param int    $index
+     */
+    private function fixWhiteSpaceAroundOperatorToAtLeastOneSingleSpace(Tokens $tokens, $index)
+    {
+        // fix white space after operator
+        if ($tokens[$index + 1]->isWhitespace()) {
+            $content = $tokens[$index + 1]->getContent();
+            if (!preg_match('/^\s+$/', $content) && false === strpos($content, "\n") && !$tokens[$tokens->getNextNonWhitespace($index + 1)]->isComment()) {
```
Comment (SpacePossum) [inline src/Fixer/Operator/BinaryOperatorSpacesFixer.php]: >if (!preg_match('/^\s+$/', $content) && false === strpos($content, "\n")

would it be possible for content to contain `"\n"` after the preg match check?

File: src/Fixer/Operator/BinaryOperatorSpacesFixer.php [hunk 2]
```diff
@@ -393,6 +405,33 @@ private function fixWhiteSpaceAroundOperatorToSingleSpace(Tokens $tokens, $index
...
+                $tokens[$index - 1] = new Token([T_WHITESPACE, ' ']);
+            }
+        } else {
+            $tokens->insertAt($index, new Token([T_WHITESPACE, ' ']));
+        }
+    }
+
     /**
      * @param Tokens $tokens
      * @param int    $index
```
Comment (SpacePossum) [inline src/Fixer/Operator/BinaryOperatorSpacesFixer.php]: needs:
- README regenerate
- unit tests
- code sample in the fixer

**Comment by AJenbo** [comment]
I would like to voice my support for this, having one space with optional alignment is ideal imo.

**Comment by VincentLanglet** [comment]
@stevelacey This option would be great !

I'm just scared about the confusion
SINGLE_SPACE -> ALIGN_SINGLE_SPACE_MINIMAL
SINGLE_SPACE_MINIMUM -> ALIGN_SINGLE_SPACE

**Comment by Wirone** [comment]
@stevelacey if you still want this to be merged, please provide tests for this feature.

**Pull request closed without merging**